pkg/pushgateway/config.go

```diff
@@ -15,17 +17,44 @@ const (
 
 // Config is the config for the template collector
 type Config struct {
+	Debug        bool
+	JobName      string
+	Labels       map[string]string
 	PushGWUrl    string
 	PushInterval time.Duration
-	JobName      string
 }
 
 // NewConfig creates a new Config instance from the provided output.Params
 func NewConfig(p output.Params) (Config, error) {
 	cfg := Config{
+		Debug:        false,
+		JobName:      defaultJobName,
+		Labels:       map[string]string{},
 		PushGWUrl:    defaultPushGWUrl,
 		PushInterval: defaultPushInterval,
-		JobName:      defaultJobName,
+	}
+
+	if _, ok := p.Environment["K6_DEBUG"]; ok {
+		cfg.Debug = true
+	}
+
+	if val, ok := p.ScriptOptions.External["pushgateway"]; ok {
+		err := json.Unmarshal(val, &cfg.Labels)
+		if err != nil {
```

Comment:
Could you consider returning an error (`fmt.Errorf("...")`) instead?

Reply:
I use `errors.Wrap`. The reason:
* it contains stack trace
* it is used by upstream libraries